### PR TITLE
Bte 130 cache

### PR DESCRIPTION
**Pull request by BruceChar**
## Context

* Describe the changes made and why they were made.
 subquery request cache!
* Provide links to any issues or designs for this change.

## Were any new external packages added?

* List external packages added and why they were added
@pacote/option @pacote/result

### CODE REVIEW

**Review by zhihuizhangjaden — APPROVED** [review]
LGTM

**Review by xsteadybcgo — APPROVED** [review]

**Comment by BruceChar** [comment]
@xsteadybcgo need your help to resolve the conflicts

File: packages/app/src/utils/subql.ts [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+      return Ok(defaultEvents)
+    }
+    if (events.startsWith('[')) {
+      events = events.slice(1, -1)
+    }
+    const evts = events.split(',')
+    evts.map((evt: string) => {
+      methods.push(methodMapping(evt))
+    })
+    return Ok(`
```
Comment (xsteadybcgo) [inline packages/app/src/utils/subql.ts:22]: Seem to be broken, does the `Ok` return the `string`? please check the endpoint in your local environment.

**Review by xsteadybcgo — CHANGES_REQUESTED** [review]

**Comment by BruceChar** [comment]
why have 2 merges log? @xsteadybcgo

**Review by xsteadybcgo — APPROVED** [review]

File: packages/app/src/utils/index.ts [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+    return result
+  }
+  query
+    .replaceAll('%20', '')
+    .split('&')
+    .forEach(para => {
+      const p = para.split('=')
+      result[p[0]] = p[1]
+    })
+  return result
```
Comment (xsteadybcgo) [inline packages/app/src/utils/index.ts:25]: I left a comment here, we need to discuss about this?

- what does the `%20` mean? 
- the `para` to `pair` ?
- what if passing a duplicate query key?

**Review by xsteadybcgo — APPROVED** [review]

**Comment by xsteadybcgo** [comment]
> why have 2 merges log? @xsteadybcgo

find some issue about the `asset_id`, i renamed it accidentally, the second one is a commit to fix it

**Pull request merged**